sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1390,6 +1546,14 @@ std::vector<node> get_root_nodes() const;
 ----
 |Returns a list of all nodes in the graph which have no dependencies.
 
+|
+[source,c++]
+----
+size_t get_mem_alloc_size() const;
+----
+|Returns the size in bytes of the physical memory allocated by this
```

Comment:
I think this should have different behaviour for a modifiable graph and a finalised graph, as the finalised graph can also incorporate optimisations.

Reply:
I think we should also have queries on the context for the amount of memory reserved and used for the whole device, something like `get_reserved_size_current` and `get_used_size_current`

Reply:
Currently this function is only available for graphs in the executable state, so it cannot be called on a modifiable graph. I think having it available for both is less relevant though. Using graph allocations creates a much more strict 1:1 relationship between an executable graph and a modifiable graph (because it can only be finalized once), and I think queries on the executable graph are more valuable because it can actually be used for execution.